src/main/js/component/common/ConfirmModal.js

```diff
@@ -0,0 +1,86 @@
+import React, { Component } from 'react';
+import { Modal } from 'react-bootstrap';
+import PropTypes from 'prop-types';
+
+class ConfirmModal extends Component {
+    constructor(props) {
+        super(props);
+        this.handleAffirmativeClick = this.handleAffirmativeClick.bind(this);
+        this.handleNegativeClick = this.handleNegativeClick.bind(this);
+        this.resetState = this.resetState.bind(this);
+        this.state = {
+            show: props.showModal,
+            confirmed: false
+        };
+    }
+
+    componentDidUpdate(prevProps) {
+        if (prevProps.showModal != this.props.showModal) {
+            this.setState({
+                show: this.props.showModal
+            });
+        }
+    }
+
+    handleAffirmativeClick(event) {
+        if (event) {
+            event.preventDefault();
+            event.stopPropagation();
+        }
+        this.props.affirmativeAction();
+        this.resetState();
+    }
+
+    handleNegativeClick(event) {
+        if (event) {
+            event.preventDefault();
+            event.stopPropagation();
+        }
+        this.props.negativeAction();
+        this.resetState();
+    }
+
+    resetState() {
+        this.setState({
+            show: false,
+            confirmed: false
+        });
+    }
+
+    render() {
+        const { affirmativeButtonText, message, negativeButtonText, title } = this.props;
+        return (
+            <Modal show={this.state.show} onHide={this.handleNegativeClick}>
+                <Modal.Header closeButton>
+                    <Modal.Title>{title}</Modal.Title>
+                </Modal.Header>
+                <Modal.Body>
+                    {message}
+                </Modal.Body>
+                <Modal.Footer>
+                    <button id="confirmNegative" type="button" className="btn btn-link" onClick={this.handleNegativeClick}>{negativeButtonText}</button>
+                    <button id="confirmAffirmative" type="button" className="btn btn-danger" onClick={this.handleAffirmativeClick}>{affirmativeButtonText}</button>
+                </Modal.Footer>
+            </Modal>
```

Comment:
Do we have a modal component that can be used effectively here?

Reply:
I think this component is exactly that.

Reply:
You're talking about Popup.js which should move into the component folder.  But that uses our config buttons.  This is to confirm an action and it has the same look and feel as LogoutConfirmation. Which is also a confirmation dialog.

Reply:
We could try building on top of the pop up button a couple different types of pop ups. But that would be for the next version